src/kafka/inflight_activation_batcher.rs

```diff
@@ -91,32 +95,24 @@ impl Reducer for InflightActivationBatcher {
         }
 
         if runtime_config.demoted_namespaces.contains(namespace) {
-            metrics::counter!(
-                "filter.forward_task_demoted_namespace",
-                "namespace" => namespace.clone(),
-                "taskname" => task_name.clone(),
-            )
-            .increment(1);
-
-            // The default demoted topic to forward tasks to is config.kafka_long_topic if not set in runtime config.
-            let topic = runtime_config
-                .demoted_topic
-                .clone()
-                .unwrap_or(self.config.kafka_long_topic.clone());
-            let delivery = self
-                .producer
-                .send(
-                    FutureRecord::<(), Vec<u8>>::to(&topic).payload(&t.activation),
-                    Timeout::After(Duration::from_millis(self.config.send_timeout_ms)),
-                )
-                .await;
-            if delivery.is_ok() {
-                metrics::counter!("filter.forward_task_demoted_namespace_success",
-                    "namespace" => namespace.clone(),
-                    "taskname" => task_name.clone(),
-                )
-                .increment(1);
-                return Ok(());
+            match tag_for_forwarding(&t.activation) {
+                Ok(Some(tagged_activation)) => {
+                    // Forward it
+                    metrics::counter!(
+                        "filter.forward_task_demoted_namespace",
+                        "namespace" => namespace.clone(),
+                        "taskname" => task_name.clone(),
+                    )
+                    .increment(1);
+                    self.forward_batch.push(tagged_activation);
+                    return Ok(());
+                }
+                Ok(None) => {
+                    // Already forwarded, fall through to add to batch
+                }
+                Err(_) => {
+                    // Decode error, fall through to add to batch to handle in upkeep
```

Comment:
We should have a metric at least for this case, so we can tell if something weird is happening.